### PR TITLE
WIP change to use resources

### PR DESCRIPTION
**Pull request by ricochet**

### CODE REVIEW

File: wit/types.wit [hunk 1]
```diff
@@ -32,14 +31,17 @@ interface types {
...
+	resource outgoing-value {
+		constructor();
+
+		from-list: static func(list<u8>) -> outgoing-value;
+		from-stream: static func(input-stream) -> outgoing-value;
+
+		write-body-sync: func(value: outgoing-value-body-sync) -> result<_, error>;
+		write-body-async: func() -> result<outgoing-value-body-async, error>;
+	}
 	type outgoing-value-body-async = output-stream;
```
Comment (ricochet) [inline wit/types.wit:43]: More work is needed here to support streams correctly

File: wit/caching.wit [hunk 2]
```diff
@@ -16,28 +16,28 @@ interface cache {
 
     // The `get` operation returns the value passed by a previous `set` for the
     // same key within the given TTL or none if there is no such value.
-    get: func(k: key) -> future-get-result;
+    get: func(k: key) -> borrow<future-get-result>;
```
Comment (Mossaka) [inline wit/caching.wit:19]: Would it make sense to return an owned `future-get-result` resource here for consuming?

File: wit/caching.wit [hunk 3]
```diff
@@ -46,14 +46,14 @@ interface cache {
     // value is updated in-place. In the common case of computing and caching a
     // value if the given key is not already in the cache, consider using
     // `get-or-set` (below) intead of separate `get` and `set` operations.
-    set: func(k: key, v: outgoing-value, TTL-ms: option<u32>) -> future-result;
+    set: func(k: key, v: outgoing-value, TTL-ms: option<u32>) -> borrow<future-result>;
```
Comment (Mossaka) [inline wit/caching.wit:49]: same comment as above

**Review by Mossaka — COMMENTED** [review]
Thanks! This is much cleaner

**Comment by Mossaka** [comment]
superseded by #33

**Pull request closed without merging**